Makefile

```diff
@@ -502,6 +506,12 @@ k8s-run-test: calico_test.created
 	    sh -c 'cp /root/.kubeadm-dind-cluster/kubectl /bin/kubectl && ls -ltr /bin/kubectl && which kubectl && cd /code/tests/k8st && \
 	           nosetests $(K8ST_TO_RUN) -v --with-xunit --xunit-file="/code/report/k8s-tests.xml" --with-timer'
 
+# Needed for Semaphore CI (where disk space is a real issue during k8s-test)
+.PHONY: remove-go-build-image
+remove-go-build-image:
+	@echo "Removing $(CALICO_BUILD) image to save space needed for testing ..."
+	@-docker rmi $(CALICO_BUILD)
+
```

Comment:
Added this flag and target to specifically address issues with `k8s-test` failing on Semaphore CI due to 10 GB disk space hard limit. 

We were already riding that 10GB limit line before this base image change. After it, the size of calico node went 2x, and it pushed us over the edge. 

I configured the Semaphore CI job for `k8s-test` specific to use this flag to remove the Calico Go Build image right before the tests start, which saves us about 1.3GB of space. 

<img width="670" alt="Screen Shot 2019-03-30 at 9 19 33 PM" src="https://user-images.githubusercontent.com/18451894/55284831-08590500-5333-11e9-93b2-fc34e0a241ed.png">

Any running the targets locally, will not have their Go build image removed (as long as they don't pass that flag).